pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -440,23 +451,127 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
+		if err != nil && !errors.IsNotFound(err) {
+			// if there is another error than the DC not being found
+			reqLogger.Error(err, "Unable to check if a DeploymentConfig exists for Postgres.", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name)
+			return reconcile.Result{}, err
+		} else if err == nil {
+			reqLogger.Info("Found a DeploymentConfig for Postgres. Deleting it.", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name)
+			err = operatorClient.Delete(context.TODO(), foundPostgresqlDeploymentConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		//#endregion
+
+		//#region DELETE Unified Push Server DeploymentConfig as we moved to Kube Deployments now
+		upsDeploymentConfigObjectMeta := metav1.ObjectMeta{
+			Namespace: instance.Namespace,
+			Name:      instance.Name, // this is the name of the DeploymentConfig we were using
+		}
+		foundUpsDeploymentConfig := &openshiftappsv1.DeploymentConfig{}
+		err = operatorClient.Get(context.TODO(), types.NamespacedName{Name: upsDeploymentConfigObjectMeta.Name, Namespace: upsDeploymentConfigObjectMeta.Namespace}, foundUpsDeploymentConfig)
+		if err != nil && !errors.IsNotFound(err) {
+			// if there is another error than the DC not being found
+			reqLogger.Error(err, "Unable to check if a DeploymentConfig exists for Unified Push Server.", "DeploymentConfig.Namespace", foundUpsDeploymentConfig.Namespace, "DeploymentConfig.Name", foundUpsDeploymentConfig.Name)
+			return reconcile.Result{}, err
+		} else if err == nil {
+			reqLogger.Info("Found a DeploymentConfig for Unified Push Server. Deleting it.", "DeploymentConfig.Namespace", foundUpsDeploymentConfig.Namespace, "DeploymentConfig.Name", foundUpsDeploymentConfig.Name)
+			err = operatorClient.Delete(context.TODO(), foundUpsDeploymentConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		//#endregion
+	}
+
+	if imageStreamResourceExists {
+		//#region DELETE OAuth Proxy ImageStream as we moved to using static image references
+		oauthProxyImageStreamObjectMeta := metav1.ObjectMeta{
+			Namespace: instance.Namespace,
+			Name:      "ups-oauth-proxy-imagestream", // this is the name of the image stream we were using
+		}
+		foundOAuthProxyImageStream := &imagev1.ImageStream{}
+		err = operatorClient.Get(context.TODO(), types.NamespacedName{Name: oauthProxyImageStreamObjectMeta.Name, Namespace: oauthProxyImageStreamObjectMeta.Namespace}, foundOAuthProxyImageStream)
+		if err != nil && !errors.IsNotFound(err) {
+			// if there is another error than the DC not being found
+			reqLogger.Error(err, "Unable to check if a ImageStream exists for OAuth Proxy.", "ImageStream.Namespace", foundOAuthProxyImageStream.Namespace, "ImageStream.Name", foundOAuthProxyImageStream.Name)
+			return reconcile.Result{}, err
```

Comment:
For the imagestreams, I'm not sure if it's worth refusing to continue with reconciliation in any case, is it? Hopefully there won't be any errors, but if there are imagestreams leftover in the namespace, they shouldn't cause any harm, and it should be enough to log the errors. What do you think?

Reply:
Ok, sounds good.